llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -47,7 +47,19 @@ def __init__(self, om_model_config: DictConfig, tokenizer: Tokenizer):
             om_model_config.pretrained_model_name_or_path,
             trust_remote_code=om_model_config.get('trust_remote_code', True),
             use_auth_token=om_model_config.get('use_auth_token', False),
-            **om_model_config.get('config_overrides', {}))
+        )
+
+        # set config overrides
+        for k, v in om_model_config.get('config_overrides', {}).items():
+            if not hasattr(config, k):
+                raise ValueError(
+                    f'config does not have attribute "{k}" to override ({k}: {v}).'
+                )
+
+            if isinstance(getattr(config, k), Dict):
```

Comment:
```suggestion
            if isinstance(getattr(config, k), Mapping):
```

might as well be permissive